src/js/program/media-controller.js

```diff
@@ -65,8 +65,8 @@ export default class MediaController extends Eventable {
         }
         // The provider has a video tag, but has not started nor preloaded
         if (this.attached && !this.setup && !this.preloaded) {
-            mediaModel.set('preloaded', true);
             provider.preload(item);
+            mediaModel.set('preloaded', true);
```

Comment:
If `provider.preload` emits events, it makes sense to update the media model first so that callbacks can react and update that state.